### PR TITLE
Add basic sync summary dialog

### PR DESCRIPTION
**Pull request by SferaDev**
### :pushpin: References

* **Issue:** Work on #42 

### :memo: Implementation

- Rename OU to Metadata in SyncDialog.

- Update TS types for synchronization.

- Add a basic sync summary with a json viewer component of the RAW response.

- Add an error handler for server issue detection.

### :art: Screenshots

image

### CODE REVIEW

File: src/logic/synchronization.ts [hunk 1]
```diff
@@ -63,23 +64,27 @@ async function exportMetadata(d2: D2, builder: ExportBuilder): Promise<Synchroni
...
+    // TODO: Update the logger
 
-    return importResult;
+    return {
+        ...importResult.data,
+        instance: {
+            id: instance.id,
+            name: instance.name,
+            url: instance.url,
+        },
```
Comment (tokland) [inline src/logic/synchronization.ts]: It's ok like this. When the number of properties grow: `_.pick(instance, ["id", "name", "url", ...])`

File: src/logic/synchronization.ts [hunk 2]
```diff
@@ -92,13 +97,13 @@ export async function startSynchronization(d2: D2, builder: SynchronizationBuild
...
     const metadataPackage = _.deepMerge({}, ...exportResults);
 
     // Phase 2: Import metadata into destination instances
     const importPromises = builder.targetInstances.map(instanceId =>
         importMetadata(d2, instanceId, metadataPackage)
     );
-    const importResults: any[] = await Promise.all(importPromises);
-    return await _.deepMerge({}, ...importResults);
+
+    return await Promise.all(importPromises);
```
Comment (tokland) [inline src/logic/synchronization.ts]: `await` can be removed

File: src/utils/synchronization.ts [hunk 3]
```diff
@@ -43,29 +43,39 @@ export async function getMetadata(d2: D2, elements: string[]): Promise<Synchroni
...
+                password: instance.password,
+            },
+            params,
+        });
+    } catch (error) {
+        // Catch response error but rethrow unknown issues
+        if (error.response) {
+            return error.response;
+        }
+        throw error;
```
Comment (tokland) [inline src/utils/synchronization.ts]: Minor. Write explicit `else` to have code in branches. Isn't the comment a bit redundant with the code?

**Comment by adrianq** [comment]
@tokland @SferaDev I created this issue https://github.com/EyeSeeTea/metadata-synchronization/issues/50 a couple of days ago to take care of the validation issues in another PR

Re your last comment, yes we would like to have something more elaborated in the short term but for now this should do the trick

**Comment by adrianq** [comment]
Merging this, for the record:
- We will take care of validation with https://github.com/EyeSeeTea/metadata-synchronization/issues/50
- We will implement a proper summary as part of https://github.com/EyeSeeTea/metadata-synchronization/issues/42

**Pull request merged**